src/components/modals/ManagePlanConstraintsModal.svelte

```diff
@@ -0,0 +1,333 @@
+    const {
+      detail: { data, newValue },
+    } = event;
+
+    if (data) {
+      selectedConstraints = {
+        ...selectedConstraints,
+        [data.id]: newValue,
+      };
+    }
+  }
+
+  async function onUpdateConstraints(selectedConstraints: Record<number, boolean>) {
+    if ($plan) {
+      const constraintPlanSpecUpdates: {
+        constraintPlanSpecIdsToDelete: number[];
+        constraintPlanSpecsToAdd: ConstraintPlanSpecInsertInput[];
+      } = Object.keys(selectedConstraints).reduce(
+        (
+          prevConstraintPlanSpecUpdates: {
+            constraintPlanSpecIdsToDelete: number[];
+            constraintPlanSpecsToAdd: ConstraintPlanSpecInsertInput[];
+          },
+          selectedConstraintId: string,
+        ) => {
+          const constraintId = parseInt(selectedConstraintId);
+          const isSelected = selectedConstraints[constraintId];
+
+          if (isSelected) {
+            return {
+              ...prevConstraintPlanSpecUpdates,
+              constraintPlanSpecsToAdd: [
+                ...prevConstraintPlanSpecUpdates.constraintPlanSpecsToAdd,
+                {
+                  constraint_id: constraintId,
+                  constraint_revision: null,
+                  enabled: true,
+                  plan_id: $planId,
+                } as ConstraintPlanSpecInsertInput,
+              ],
+            };
+          } else {
+            return {
+              ...prevConstraintPlanSpecUpdates,
+              constraintPlanSpecIdsToDelete: [
+                ...prevConstraintPlanSpecUpdates.constraintPlanSpecIdsToDelete,
+                constraintId,
+              ],
+            };
+          }
+        },
+        {
+          constraintPlanSpecIdsToDelete: [],
+          constraintPlanSpecsToAdd: [],
+        },
+      );
+
+      await effects.updateConstraintPlanSpecifications(
+        $plan,
+        constraintPlanSpecUpdates.constraintPlanSpecsToAdd,
+        constraintPlanSpecUpdates.constraintPlanSpecIdsToDelete,
+        user,
+      );
+      dispatch('close');
+    }
+  }
+</script>
+
+<Modal height={500} width={600}>
+  <ModalHeader on:close>Manage Constraints</ModalHeader>
+  <ModalContent style="padding:0">
+    <div class="constraints-modal-container">
+      <div class="constraints-modal-filter-container">
+        <div class="constraints-modal-title">Constraints</div>
+        <Input>
+          <input bind:value={filterText} class="st-input" placeholder="Filter constraints" style="width: 100%;" />
+        </Input>
+        <button
+          class="st-button secondary ellipsis"
+          name="new-constraint"
+          use:permissionHandler={{
+            hasPermission: hasCreatePermission,
+            permissionError,
+          }}
+          on:click={() => window.open(`${base}/constraints/new?${SearchParameters.MODEL_ID}=${$plan?.model_id}`)}
+        >
+          New
+        </button>
+      </div>
+      <hr />
```

Comment:
Could the color <hr> match the border color found in the tables? Stands out a lot as is.